src/backend/marsha/core/serializers/base.py

```diff
@@ -139,6 +140,32 @@ def to_internal_value(self, value):
             raise ValidationError(error) from error
 
 
+class ReadWritePrimaryKeyRelatedField(serializers.PrimaryKeyRelatedField):
+    """A serializer field to serialize/deserialize complete related object."""
+
+    def __init__(self, to_representation_serializer, **kwargs):
+        self.to_representation_serializer = to_representation_serializer
+        super().__init__(**kwargs)
+
+    def use_pk_only_optimization(self):
+        return False
+
+    def to_internal_value(self, data):
+        # If the provided data comes from the serialization result, we can still accept the data
+        # by fetching the object id.
+        if isinstance(data, dict):
+            return super().to_internal_value(data["id"])
+        return super().to_internal_value(data)
+
+    def run_validation(self, data=empty):
+        if isinstance(data, dict):
+            return super().run_validation(data["id"])
+        return super().run_validation(data)
+
+    def to_representation(self, value):
+        return self.to_representation_serializer(value).data
+
+
```

Comment:
Even if the docstring is inherited, I suggest to add some:
```suggestion
class ReadWritePrimaryKeyRelatedField(serializers.PrimaryKeyRelatedField):
    """
    A serializer field to serialize/deserialize complete related object.

    It allows to serialize related foreign related object (many to one) for display purpose,
    while still allowing to use only its PK when editing (create or update).

    Examples (for the parent object, with this serializer for playlist):
     - GET will return
       ```
        {
            "id": "1",
            "title": "foo",
            "playlist": {
                "id": "1",
                "title": "bar",
                "lti_id": "course-v1:ufr+mathematics+00001"
            }
        }
       ```
     - while you can POST/PUT/PATCH with
       ```
        {
            "id": "1",
            "title": "foo",
            "playlist": 1
        }
       ```
    """

    def __init__(self, to_representation_serializer, **kwargs):
        """Init the serializer providing a serializer to use for the related object."""
        self.to_representation_serializer = to_representation_serializer
        super().__init__(**kwargs)

    def use_pk_only_optimization(self):
        """
        Do not use this optimization since we need to retrieve more data from the related object.
        """
        return False

    def to_internal_value(self, data):
        """
        Allow to provide only the PK of the related object (nominal case) or extract the PK
        if the POSTed data are from a previous GET, giving all the object.
        
        This allows to provide input for the parent object like
        ```
        {
            "id": "1",
            "title": "foo",
            "playlist": 1
        }
       ```
       
       or
       ```
        {
            "id": "1",
            "title": "foo",
            "playlist": {
                "id": "1",
                "title": "bar",
                "lti_id": "course-v1:ufr+mathematics+00001"
            }
        }
       ```
        """
        # If the provided data comes from the serialization result, we can still accept the data
        # by fetching the object id.
        if isinstance(data, dict):
            return super().to_internal_value(data["id"])
        return super().to_internal_value(data)

    def run_validation(self, data=empty):
        """Validate data according to the same principle as the `to_internal_value` method."""
        if isinstance(data, dict):
            return super().run_validation(data["id"])
        return super().run_validation(data)

    def to_representation(self, value):
        """Use the provided serializer to serialize the related object."""
        return self.to_representation_serializer(value).data


```